internal/services/cosmos/cosmosdb_account_resource_test.go

```diff
@@ -3452,6 +3460,10 @@ resource "azurerm_cosmosdb_account" "test" {
   backup {
     type = "Continuous"
   }
+
+  lifecycle {
+    ignore_changes = [backup.0.tier]
+  }
```

Comment:
how come this is required?

Reply:
There are two reasons.
Reason 1: Though `tier` has the default value `Continuous7Days` but [`tier`](https://github.com/Azure/azure-rest-api-specs/blob/68030f1c17e0cd5912e1bda17a5f28473e84b5f9/specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2023-04-15/cosmos-db.json#L9891) is only for the backup type `Continuous`. So the default value isn't added in the property schema.
Reason 2: API would return the default value `Continuous7Days` when backup.type is `Continuous` and `tier` isn't set. At this time, it would cause diff. So here ignore_changes for `tier` is added.

Note: I updated PR to add the note for it.

Reply:
I feel like this is a bug, is there no way to correctly handle this so lifecycle ignore changes is not requireD? is this an API bug with the API not returning the correct values?

Reply:
Service team confirmed that it's by API design. Actually, This new property is called [ContinuousTier](https://github.com/Azure/azure-rest-api-specs/blob/68030f1c17e0cd5912e1bda17a5f28473e84b5f9/specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2023-04-15/cosmos-db.json#L9891) in azure-rest-api-spec, which means it only supports the backup type ["Continuous"](https://github.com/hashicorp/terraform-provider-azurerm/blob/11d2d5e59bd61b78d36e5cce8d60b51d6d41fa92/internal/services/cosmos/cosmosdb_account_resource.go#L525). So I assume that it's not an API bug. I have listed the behaviors for this property.

Scenario 1: There is no default value for this new property "ContinuousTier" when [backup type](https://github.com/hashicorp/terraform-provider-azurerm/blob/11d2d5e59bd61b78d36e5cce8d60b51d6d41fa92/internal/services/cosmos/cosmosdb_account_resource.go#L521) is "Periodic" and "ContinuousTier" isn't set.
Scenario 2: API doesn't return any value for this new property "ContinuousTier" when [backup type](https://github.com/hashicorp/terraform-provider-azurerm/blob/11d2d5e59bd61b78d36e5cce8d60b51d6d41fa92/internal/services/cosmos/cosmosdb_account_resource.go#L521) is "Periodic" and "ContinuousTier" is set.
Scenario 3: API returns default value "Continuous7Days" for this new property "ContinuousTier" when [backup type](https://github.com/hashicorp/terraform-provider-azurerm/blob/11d2d5e59bd61b78d36e5cce8d60b51d6d41fa92/internal/services/cosmos/cosmosdb_account_resource.go#L521) is "Continuous" and "ContinuousTier" isn't set.
Scenario 4: API returns value "Continuous7Days" for this new property "ContinuousTier" when [backup type](https://github.com/hashicorp/terraform-provider-azurerm/blob/11d2d5e59bd61b78d36e5cce8d60b51d6d41fa92/internal/services/cosmos/cosmosdb_account_resource.go#L521) is "Continuous" and ContinuousTier is set to "Continuous7Days".
Scenario 5: API returns value "Continuous30Days" for this new property "ContinuousTier" when [backup type](https://github.com/hashicorp/terraform-provider-azurerm/blob/11d2d5e59bd61b78d36e5cce8d60b51d6d41fa92/internal/services/cosmos/cosmosdb_account_resource.go#L521) is "Continuous" and ContinuousTier is set to "Continuous30Days".